app/src/main/java/com/example/stromtracker/ui/urlaub/urlaub_edit/UrlaubEditFragment.kt

```diff
@@ -0,0 +1,192 @@
+    private lateinit var fragMan: FragmentManager
+
+    private lateinit var name: EditText
+    private lateinit var start: EditText
+    private lateinit var ende: EditText
+
+    private lateinit var outTage: TextView
+    private lateinit var outErsparnis: TextView
+
+    private lateinit var btnAbbrechen: Button
+    private lateinit var btnSpeichern: Button
+    private lateinit var btnLoeschen: Button
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        urlaubViewModel = ViewModelProvider(this).get(UrlaubViewModel::class.java)
+
+        val root = inflater.inflate(R.layout.fragment_urlaub_edit, container, false)
+        fragMan = parentFragmentManager
+
+        outTage = root.findViewById(R.id.urlaub_edit_text_tage)
+        outErsparnis = root.findViewById(R.id.urlaub_edit_text_ersparnis)
+        name = root.findViewById(R.id.urlaub_edit_edit_name)
+        start = root.findViewById(R.id.urlaub_edit_edit_datum_von)
+        addCustomTextChangedListener(start)
+        ende = root.findViewById(R.id.urlaub_edit_edit_datum_bis)
+        addCustomTextChangedListener(ende)
+
+        name.setText(urlaub.getName())
+        var tempStr = SimpleDateFormat("dd.MM.yyyy", Locale.GERMAN).format(urlaub.getDateVon())
+        start.setText(tempStr)
+        tempStr = SimpleDateFormat("dd.MM.yyyy", Locale.GERMAN).format(urlaub.getDateBis())
+        ende.setText(tempStr)
+
+        btnAbbrechen = root.findViewById(R.id.urlaub_edit_button_abbrechen)
+        btnAbbrechen.setOnClickListener(this)
+        btnSpeichern = root.findViewById(R.id.urlaub_edit_button_speichern)
+        btnSpeichern.setOnClickListener(this)
+        btnLoeschen = root.findViewById(R.id.urlaub_edit_button_delete)
+        btnLoeschen.setOnClickListener(this)
+
+        return root
+    }
+
+
+    private fun addCustomTextChangedListener(edit: EditText): EditText {
+        edit.addTextChangedListener(object : TextWatcher {
+            override fun afterTextChanged(s: Editable) {
+                if (checkDates(start, ende)) {
+                    val start =
+                        SimpleDateFormat(
+                            "dd.MM.yyyy",
+                            Locale.GERMAN
+                        ).parse(start.text.toString())
+                    val ende =
+                        SimpleDateFormat(
+                            "dd.MM.yyyy",
+                            Locale.GERMAN
+                        ).parse(ende.text.toString())
+                    updateOutputs(start, ende)
+                }
+            }
+
+            override fun beforeTextChanged(
+                s: CharSequence,
+                start: Int,
+                count: Int,
+                after: Int
+            ) {
+            }
+
+            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {}
+        })
+        return edit
+    }
```

Comment:
Warum machst du hier einen TextListener

Reply:
Der TextChangedListener ist dafür da um die Anzeige für die Dauer in Tagen und die daraus resultierende Ersparnis live anzuzuzeigen. 
Er triggert jedes mal, wenn eines der Datums geändert wird

Reply:
Ok gut verstehe